src/core/services/interimElement/interimElement.js

```diff
@@ -323,30 +329,40 @@ function InterimElementProvider() {
        */
       function hide(reason, options) {
         if ( !stack.length ) return $q.when(reason);
+        if (hideQueue >= stack.length) return;
         options = options || {};
 
         if (options.closeAll) {
-          var promise = $q.all(stack.reverse().map(closeElement));
-          stack = [];
-          return promise;
+          return $q.all(stack.reverse().map(closeElement));
         } else if (options.closeTo !== undefined) {
-          return $q.all(stack.splice(options.closeTo).map(closeElement));
+          return $q.all(stack.slice(options.closeTo).map(closeElement));
```

Comment:
Yes, that's by intention, because we don't want to remove items from stack when the close animation starts. They will be removed later, after the animation is done.